### PR TITLE
Adding nectarchain package

### PR DESCRIPTION
**Pull request by jlenain**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jlenain** [comment]
@conda-forge/help-python, ready for review!

**Comment by jlenain** [comment]
@conda-forge/help-python, ready for review!

**Comment by jlenain** [comment]
Will wait until `ctapipe_io_nectarcam` enters `conda-forge` through #22372 , since `nectarchain` depends on it.

**Comment by jlenain** [comment]
Hi @conda-forge-admin , please ping conda-forge/help-python. It is ready for review.

File: recipes/nectarchain/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+requirements:
+  host:
+    - python =3
+    - setuptools >=64.0.3
+    - setuptools-scm >=3.4
+    - pip
+  run:
+    - python =3
+    - ctapipe =0.12
+    - ctapipe-io-nectarcam
```
Comment (dopplershift) [inline recipes/nectarchain/meta.yaml:26]: Any reason this doesn't match upstream, which has:
```
    astropy~=4.2
    ctapipe~=0.12
    numpy~=1.22
    protozfits~=2.0
    tables>=3.7
```
?

Comment (jlenain) [reply recipes/nectarchain/meta.yaml:26]: Ah, my mistake.

Comment (jlenain) [reply recipes/nectarchain/meta.yaml:27]: The last commit (3eb328064acc251e89a8c28490fa3360116b805f) introduces the dependencies from upstream (e.g. on `astropy`, `ctapipe`, or `protozfits`), with light constraints as in #22372 , on which the presently proposed package depends.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**